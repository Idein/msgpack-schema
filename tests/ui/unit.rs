use msgpack_schema::*;

mod serialize {
    use super::*;

    #[derive(Serialize)]
    struct S1;

    #[derive(Serialize)]
    struct S3();
}

mod deserialize {
    use super::*;

    #[derive(Deserialize)]
    struct S2;

    #[derive(Deserialize)]
    struct S4();
}

fn main() {}
