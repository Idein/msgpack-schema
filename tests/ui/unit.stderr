error: unit structs as serialize are not supported
 --> $DIR/unit.rs:7:5
  |
7 |     struct S1;
  |     ^^^^^^^^^^

error: empty tuple structs as serialize are not supported
  --> $DIR/unit.rs:10:5
   |
10 |     struct S3();
   |     ^^^^^^^^^^^^

error: unit structs as deserialize are not supported
  --> $DIR/unit.rs:17:5
   |
17 |     struct S2;
   |     ^^^^^^^^^^

error: empty tuple structs as deserialize are not supported
  --> $DIR/unit.rs:20:5
   |
20 |     struct S4();
   |     ^^^^^^^^^^^^
